hal/armv8m/exceptions.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system kernel
+ *
+ * Exception handling
+ *
+ * Copyright 2017, 2022 Phoenix Systems
+ * Author: Pawel Pisarczyk, Jakub Sejdak, Damian Loewnau
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "../exceptions.h"
+#include "../cpu.h"
+#include "../console.h"
+#include "../string.h"
+#include "config.h"
+
+
+void hal_exceptionsDumpContext(char *buff, exc_context_t *ctx, int n)
+{
+	static const char *mnemonics[] = {
+		"0 #InitialSP", "1 #Reset", "2 #NMI", "3 #HardFault",
+		"4 #MemMgtFault", "5 #BusFault", "6 #UsageFault", "7 #SecureFault",
+		"8 #", "9 #", "10 #", "11 #SVC",
+		"12 #Debug", "13 #", "14 #PendSV", "15 #SysTick"
+	};
+	size_t i = 0;
+
+	n &= 0xf;
+
+	hal_strcpy(buff, "\nException: ");
+	hal_strcpy(buff += hal_strlen(buff), mnemonics[n]);
+	hal_strcpy(buff += hal_strlen(buff), "\n");
+	buff += hal_strlen(buff);
+
+	i += hal_i2s(" r0=", &buff[i], ctx->r0, 16, 1);
```

Comment:
```suggestion
	hal_strcpy(buff, "\nException: ");
	i = sizeof("\nException: ") - 1;
	hal_strcpy(&buff[i], mnemonics[n]);
	i += hal_strlen(mnemonics[n]);

	i += hal_i2s("\n r0=", &buff[i], ctx->r0, 16, 1);
```
`hal_strlen()` only on `mnemonics[n]`, without 3 times on  "\nExceptions: "

General comment.
Since the `hal_i2s()` function is called with afixed 4th and 5th argument, you can reduce code size by defining, for example
```
hal_i2sHex(const char *prefix, char *buff, unsigned long i);
```
and use it.

You can define this function as:
```
size_t hal_exceptionsDumpContext(char *buff, exc_context_t *ctx, int n)
```
and eliminate `hal_strlen()` in this code:
https://github.com/phoenix-rtos/phoenix-rtos-kernel/blob/428d509d25d24967a4d88235effe0d7390a6677a/proc/process.c#L324-L330

Reply:
Thanks! I think changing `hal_exceptionsDumpContext()` definition can be very good idea, but maybe for another PR.